matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/permalinks/PermalinkFactory.kt

```diff
@@ -55,14 +73,26 @@ internal class PermalinkFactory @Inject constructor(
         }
     }
 
-    fun createPermalink(roomId: String, eventId: String): String {
-        return MATRIX_TO_URL_BASE + escape(roomId) + "/" + escape(eventId) + viaParameterFinder.computeViaParams(userId, roomId)
+    fun createPermalink(roomId: String, eventId: String, forceMatrixTo: Boolean): String {
+        return buildString {
+            append(baseUrl(forceMatrixTo))
+            if (useClientFormat(forceMatrixTo)) {
+                append(ROOM_PATH)
+            }
+            append(escape(roomId))
+            append("/")
+            append(escape(eventId))
+            append(viaParameterFinder.computeViaParams(userId, roomId))
+        }
     }
 
     fun getLinkedId(url: String): String? {
-        return url
-                .takeIf { it.startsWith(MATRIX_TO_URL_BASE) }
-                ?.substring(MATRIX_TO_URL_BASE.length)
+        val clientBaseUrl = matrixConfiguration.clientPermalinkBaseUrl
+        return when {
+            url.startsWith(MATRIX_TO_URL_BASE)                     -> url.substring(MATRIX_TO_URL_BASE.length)
+            clientBaseUrl != null && url.startsWith(clientBaseUrl) -> url.substring(clientBaseUrl.length)
```

Comment:
You will have to remove the specific path (defined [here](https://github.com/tchapgouv/tchap-android-v2/pull/162/commits/adb678e46a1610e1a513eec6a88bee17dfc252fc#diff-ebd7145fe965a4e4c0dbac343fd963ffa2c88e5aaa8e77f0c7789288b6edd217R137)), before being able to extract the id  with `substringBeforeLast("?")`